### DISCUSSION TITLE
Problems with the ampersand (&) special character

### DISCUSSION THREAD

**Discussion by AndroSavvy** [chrisant996/clink#623] [General]
I am using _Clink v1.6.16_ with Windows Command Prompt and an Oh-My-Posh theme and I am not able to  `cd` to a directory name containing an ampersand **&**. 
```batchfile
cd "_Stores&Orders"
Orders' is not recognized as an internal or external command, operable program or batch file.
``` 
If I use the escape character `^` before **&** I get another error:
```batchfile
cd "_Stores^&Orders"
The system cannot find the path specified.
```
But if I exit Clink using `cmd /D` the `cd` is working as it should.

Any idea what the problem is?

**Comment by chrisant996** [comment]
I think whatever you are encountering is external to Clink.
It works fine for me:

![image](https://github.com/chrisant996/clink/assets/17440311/a9310eb3-5a55-43a6-b3f7-094ef13c9d37)

Do you use any kind of enhanced directory changing tool such as [z.lua](https://github.com/skywind3000/z.lua) or [zoxide](https://github.com/ajeetdsouza/zoxide) or [CD Deluxe](https://github.com/m6z/cd-deluxe) or etc?

I bet you have an alias and/or batch script for `cd`, and the script doesn't handle quoted arguments quite correctly.

**Reply by AndroSavvy** [reply]
You are right Christian. My bad. Sorry about that!
It's indeed a problem with oh-my-posh. If I disable the **omp** theme in `oh-my-posh.lua` from the **Clink** directory `%LOCALAPPDATA%\clink\`. I tried different themes and the same problem occurs. It's strange, as in PowerShell and Git Bash, where I use as well **omp** themes, this problem doesn't occur. As I'm a total newb, I have no idea what and where to look for in the theme file. I'll search on omp discussion threads to see if there is a solution.

P.S. _If not asking too much, what theme are you using?_

**Reply by chrisant996** [reply]
I don't use oh-my-posh.
I use [clink-flex-prompt](https://github.com/chrisant996/clink-flex-prompt).

But the `cd` issue is not related to oh-my-posh.

What is the output from running `where cd` ?

**Reply by AndroSavvy** [reply]
```batchfile
where cd
INFO: Could not find files for the given pattern(s).
```
Did you mean `where cmd`?
```batchfile
where cmd
C:\Windows\System32\cmd.exe
```

**Reply by chrisant996** [reply]
> Did you mean `where cmd`?

No, I meant `cd`.

You're experiencing problems with `cd`.

It seems like you've got a doskey alias that's replacing the `cd` command.

**Reply by chrisant996** [reply]
@AndroSavvy what happens if you put an extra space character before the `cd` command? 

`   cd "_Stores&Orders"`

The extra space(s) bypasses any aliases and invokes the built-in CMD `cd` command directly.

**Reply by AndroSavvy** [reply]
This was the problem! 
I had an `Aliases.cmd` file with some `doskey` lines, one of them being `doskey pwd=echo %cd%`.
I commented out the line and now the `cd` is working again as it should.

Thank you very much Christian for helping me solve the problem!

**Reply by chrisant996** [reply]
Oh, that's very interesting.

This seems to be related to oh-my-posh after all.
Apparently oh-my-posh must be looking up an alias defined for `pwd` and then running it???
But that seems very strange.

- What version of oh-my-posh are you using?
- Can you share your `oh-my-posh.lua` script?

Also, the `pwd` alias has a bug and will always malfunction with a directory whose name contains `&`.  Because the command `echo %cd%` doesn't do anything to add quotes when the directory name contains `&`.  So it becomes `echo C:\tmp\_Stores&Orders` which means run `echo C:\tmp\_Stores` and then run `Orders`.

**Reply by AndroSavvy** [reply]
No problem.

oh-my-posh.lua
```lua
load(io.popen('oh-my-posh init cmd --config C:/Users/<UserName>/AppData/Local/Programs/oh-my-posh/themes/takuya_cust.omp.json'):read("*a"))()
``` 
```batchfile 
oh-my-posh version
21.1.0
```
I learned that instead of `pwd` I can use anyhow `cd` to see the current directory path.
